llvm/lib/Transforms/InstCombine/InstCombineAndOrXor.cpp

```diff
@@ -4685,6 +4685,24 @@ Instruction *InstCombinerImpl::foldNot(BinaryOperator &I) {
   return nullptr;
 }
 
+static Instruction *sinkNotIntoBitwiseOp(BinaryOperator &I) {
+  if (!match(&I, m_Not(m_Value())))
+    return nullptr;
+  BasicBlock *BB = I.getParent();
+  if (!all_of(I.users(), [BB](const User *U) {
+        auto *UI = dyn_cast<Instruction>(U);
+        // TODO: check if target has ANDN/ORN/XNOR
```

Comment:
InstCombine can't check this. https://llvm.org/docs/InstCombineContributorGuide.html#canonicalization-and-target-independence